### PR TITLE
C2 karimngr c1s15

### PR DESCRIPTION
**Pull request by ghost**
@AmrARaouf  @Mirna-Yacout  @Omar-Hossam  @m0hamed

### CODE REVIEW

**Comment by Mirna-Yacout** [comment]
UML reviewed

**Comment by ghost** [comment]
files changed 
/app/controllers/admin_controller.rb
/app/models/trophy.rb
/app/views/admin/index.html.erb
/app/views/admin/_admin-add-trophy.html.erb

File: arability/app/models/trophy.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                            only_integer: true, 
+                            greater_than: 0,
+                            less_than_or_equal_to: 10000,
+                            message: "مجموع النقاط يجب ان يكون بين 0 و 10000"
+
+  validates_attachment_content_type :image, 
+                                    :content_type => /^image\/(png|gif|jpeg)/,
+                                    message: "الصورة يجب ان تكون بصيغة png, gif او jpeg"
+
+  class << self
```
Comment (omar-mohyeldin) [inline arability/app/models/trophy.rb]: avoid class << self. Use self.method instead

File: arability/app/controllers/admin_controller.rb [hunk 2]
```diff
@@ -40,109 +45,179 @@ def check_login
...
   # failure: 
-  #     none
-  def create_session
-    session[:who_is_this] = "admin"
+  #     refreshes the page with error displayed
+  def add_trophy
+    params[:name] = params[:name].strip
+    params[:level] = params[:level].strip
+    params[:score] = params[:score].strip
+    success, trophy = Trophy.add_trophy_to_database(params[:name], params[:level], params[:score], params[:image])
```
Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb]: please divide your line, you are allowed 80 characters maximum per line

File: arability/app/controllers/admin_controller.rb [hunk 3]
```diff
@@ -40,109 +45,179 @@ def check_login
...
+      flash[:successtype] = "addtrophy"
+    else
+      flash[:error] = trophy.errors.messages
+      flash[:errortype] = "addtrophy"
+    end
+    flash.keep
+    if success
+      redirect_to action: "index", anchor: "admin-list-trophies"
+    else
+      redirect_to action: "index", anchor: "admin-add-trophy", fargs: {addtrophy: params}
```
Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb]: please divide your line, you are allowed 80 characters maximum per line

File: arability/app/controllers/admin_controller.rb [hunk 4]
```diff
@@ -40,109 +45,179 @@ def check_login
...
-    session[:who_is_this] = nil
-  end
-
-  def index
+  #     refreshes the page with error displayed
+  def add_prize
+    params[:name] = params[:name].strip
+    params[:level] = params[:level].strip
+    params[:score] = params[:score].strip
+    success, prize = Prize.add_prize_to_database(params[:name], params[:level], params[:score], params[:image])
```
Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb]: please divide your line, you are allowed 80 characters maximum per line

File: arability/app/controllers/admin_controller.rb [hunk 5]
```diff
@@ -40,109 +45,179 @@ def check_login
...
+      flash[:successtype] = "addprize"
+    else
+      flash[:error] = prize.errors.messages
+      flash[:errortype] = "addprize"
+    end
+    flash.keep
+    if success
+      redirect_to action: "index", anchor: "admin-list-prizes"
+    else
+      redirect_to action: "index", anchor: "admin-add-prize", fargs: {addprize: params}
```
Comment (omar-mohyeldin) [inline arability/app/controllers/admin_controller.rb]: please divide your line, you are allowed 80 characters maximum per line

File: arability/app/models/trophy.rb [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    #     name: the name of the trophy
+    #     level: the level of the trophy
+    #     score: the score of the trophy 
+    #     image: the image of the trophy
+    # success: 
+    #     returns true and the new trophy if it is added to the database
+    # failure: 
+    #     returns false and the trophy if it is not added to the database
+    def add_trophy_to_database(name, level, score, image)
+      new_trophy = Trophy.new(name: name, level: level, score: score, image: image)
```
Comment (omar-mohyeldin) [inline arability/app/models/trophy.rb:63]: please divide your line, you are allowed a maximum of 80 characters per line

**Comment by omar-mohyeldin** [comment]
@karimngr please modify your code

**Comment by ghost** [comment]
the issue is #173

**Comment by ghost** [comment]
@Omar-Hossam

**Comment by omar-mohyeldin** [comment]
@karimngr code verified

**Comment by Mirna-Yacout** [comment]
UML reviewed

**Pull request merged**